src/Resources/web/app_dev.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+use Contao\ManagerBundle\ContaoManager\Plugin as ManagerBundlePlugin;
+use Contao\ManagerBundle\HttpKernel\ContaoKernel;
+use Doctrine\Common\Annotations\AnnotationRegistry;
+use Symfony\Component\Debug\Debug;
+use Symfony\Component\Dotenv\Dotenv;
+use Symfony\Component\HttpFoundation\Request;
+
+// The check is to ensure we don't use .env in production
```

Comment:
While I do not really get what you mean with this comment (the check is in `app_dev.php` and has nothing to do with `app.php`), it feels wrong to me with respect to future compat with symfony flex and also with current implementation.
This causes not only this feature to only work in dev mode but also all other `getenv` stuff.
IMO we should use `dotenv` in both endpoints, `app.php` and `app_dev.php` to make behaviour consistent.
You are right however about not adding the auth check to `app.php`.

So please also add this also to `app.php`